book/src/tooling/coap.md

```diff
@@ -0,0 +1,256 @@
+which can then be used for fast communication.
+
+Unless ACE or certificate chains are used,
+the main use of EDHOC in RIOT-rs is with raw public keys:
+Devices (including the host machine) generate a private key,
+make the corresponding public key known,
+and then send the public key (or its short ID) along with EDHOC messages to be identified by that public key.
+This is similar to how SSH keys are commonly used.
+
+Policies described in terms of EDHOC keys include the public key of the peer,
+which private key to use,
+and whether our public key needs to be sent as a full public key or can be sent by key ID.
+
+[RFC9528]: https://datatracker.ietf.org/doc/html/rfc9528
+
+### ACE
+
+The ACE framework ([RFC9200]) describes how a trusted service (the Authorization Server, "AS")
+can facilitate secure connections between devices that are not explicitly configured to be used together.
+It frequently issues CWTs (CBOR Web Tokens) that are to JWTs (JSON Web Tokens) as CoAP is to HTTP.
+
+The general theme of it is that
+a client that wants to access some resource on a server under the AS's control
+first asks the AS for an access token,
+and then presents that token to the server (called Resource Server / RS in that context).
+
+Clients know the AS by having means to establish some secure connection with it,
+eg. through EDHOC.
+Resource Servers know the AS by verifying the tokens they receive from the AS,
+eg. by having a shared secret with the AS or knowing its signing key.
+
+Details of the process are specified in ACE profiles;
+apart from those listed here, popular profiles include the DTLS profile and the profiles for group communication.
+
+[RFC9200]: https://datatracker.ietf.org/doc/html/rfc9200
+
+#### ACE-OSCORE profile
+
+With the ACE-OSCORE profile ([RFC9203]),
+the AS provides a random OSCORE key in the token (which is encrypted for the RS),
+and sends the token along with the same OSCORE key through its secure connection to the client.
+Before the client can send OSCORE requests,
+it POSTs the token to the server over an unprotected connection
+(the token itself is encrypted),
+along with a random number and some housekeeping
+that go into the establishment of an OSCORE context.
+
+The [documentation of the CoAP/ACE-OAuth PoC project] describes the whole setup in more detail.
+
+[documentation of the CoAP/ACE-OAuth PoC project]: https://gitlab.com/oscore/coap-ace-poc-overview
+
+[RFC9203]: https://datatracker.ietf.org/doc/html/rfc9203
+
+#### ACE-EDHOC profile
+
+The ACE-EDHOC profile is [under development in the ACE working group].
+
+It differs from the ACE-OSCORE profile
+in that the AS does not provide symmetric key material
+but only points out the respective peer's public keys.
+
+The token is not sent in plain,
+but as part of the EDHOC exchange.
+
+[under development in the ACE working group]: https://datatracker.ietf.org/doc/draft-ietf-ace-edhoc-oscore-profile/
+
+#### ACE during development
```

Comment:
you mean "under development"?

Reply:
That's just the link expansion. After rendering, this reads as

> The ACE-EDHOC profile is [under development in the ACE working group].
>
> [under development in the ACE working group]: https://datatracker.ietf.org/doc/draft-ietf-ace-edhoc-oscore-profile/

(which I think is the right semantic span to place the link under, following the spirit of [Don't Click Here](https://www.w3.org/QA/Tips/noClickHere) and making the span a sensible title for the target)

Reply:
My comment was about the title of the section... should it not be "ACE under development" ?

Reply:
Ah, that's what you mean:

No, this refers to the first two paragraphs of that -- you *can* have a full-blown ACE Authorization Server, but here we describe how ACE is used in a minimal local-machine mode.